sklearn/linear_model/_ridge.py

```diff
@@ -1502,7 +1502,10 @@ def fit(self, X, y, sample_weight=None):
                 alpha_score = scorer(
                     _IdentityEstimator(), predictions.ravel(), y.ravel())
                 if self.store_cv_values:
-                    self.cv_values_[:, i] = predictions.ravel()
+                    # compute the predictions in the original space
+                    self.cv_values_[:, i] = (
+                        predictions.ravel() / np.tile(sqrt_sw, n_y)
```

Comment:
it may not be a very meaningful input but I don't think anything forbids some sample weights to be 0
```python
import numpy as np
from sklearn.linear_model import RidgeCV
from sklearn.datasets import make_regression

x, y = make_regression()
sw = np.ones(len(x))
sw[0] = 0.0
r = RidgeCV(store_cv_values=True, scoring="neg_mean_squared_error").fit(x, y, sample_weight=sw)
print(r.cv_values_.mean(axis=0)) # nan
```

Reply:
What would you do in this case. You cannot recover predictions in the original space, isn't it?

Reply:
yes I guess we cannot recover the predictions but at least with the change you made we avoid the division by 0 warning

Reply:
as rescaling multiplies by the square root of sample weights the prediction would indeed be the intercept for those samples